### PR TITLE
Backports for 3.4.2

### PR DESCRIPTION
**Pull request by mattpap**
This PR collects bug-fixes and tasks for a 3.4.2 release:

- [x] https://github.com/bokeh/bokeh/pull/13832
- [x] https://github.com/bokeh/bokeh/pull/13918
- [x] https://github.com/bokeh/bokeh/pull/13840 (partial; only mypy related changes)
- [x] https://github.com/bokeh/bokeh/pull/13911 
- [x] https://github.com/bokeh/bokeh/pull/13843
- [x] https://github.com/bokeh/bokeh/pull/13851
- [x] 3.4.2 release notes
- [x] updated switcher.json

Release checklist:

- [x] do **not** squash merge
- [x] update milestone to `3.4.2` of relevant issues and PRs
- [x] remove `NEEDS BACKPORT` label

### CODE REVIEW

**Comment by mattpap** [comment]
This is tentatively ready, but I will wait with this for 3.5.0rc1, to make sure we aren't missing anything last minute.

**Comment by bryevdv** [comment]
3.4.1 can be removed from the switcher. Intended policy to date has been to only include the latest point release for each minor release level

**Comment by mosc9575** [comment]
Sorry for the late reaction. 

Ithink it would be good to add

- [x] https://github.com/bokeh/bokeh/pull/13823
- [x] https://github.com/bokeh/bokeh/pull/13876

And  instead of changing the switcher.json by hand we could enable the version banner (including an update for ther switcher.json).

- [x] https://github.com/bokeh/bokeh/pull/13901
- [x] https://github.com/bokeh/bokeh/pull/13903

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
LGTM @mosc9575 can you elaborate on
> And instead of changing the switcher.json by hand we could enable the version banner (including an update for ther switcher.json).

even with the new banner we still have to update the switcher with the correct version information each release AFAIK

**Comment by mosc9575** [comment]
> LGTM @mosc9575 can you elaborate on
> 
> > And instead of changing the switcher.json by hand we could enable the version banner (including an update for ther switcher.json).
> 
> even with the new banner we still have to update the switcher with the correct version information each release AFAIK

Your are totally right.

File: docs/bokeh/switcher.json [hunk 1]
```diff
@@ -1,15 +1,9 @@
...
-  },
-  {
-    "version": "3.4.0",
-    "url": "https://docs.bokeh.org/en/3.4.0/"
+    "version": "3.4.2",
+    "url": "https://docs.bokeh.org/en/latest/",
+    "preferred": true
   },
   {
     "version": "3.3.0",
```
Comment (mosc9575) [inline docs/bokeh/switcher.json:9]:
```suggestion
    "version": "3.3.4",
```

The last minor version in the 3.3 series is `3.3.4`. The same issue exists in branch 3.5. I don't know if it is worth to update this line.

**Review by mosc9575 — APPROVED** [review]

**Comment by bryevdv** [comment]
Tangentially related: all 3.4.x point release notes are missing from `branch-3.5`

**Pull request merged**